pkg/serviceaccounts/oauthclient/oauthclientregistry.go

```diff
@@ -0,0 +1,109 @@
+package oauthclient
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+
+	scopeauthorizer "github.com/openshift/origin/pkg/authorization/authorizer/scope"
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/oauthclient"
+)
+
+const (
+	OAuthClientSecretAnnotation            = "serviceaccounts.openshift.io/oauth-client-secret"
```

Comment:
> Should this be oauth.openshift.io/client-secret? Is this oauth API or service account API?

It only makes sense to specify this value on an SA token, so I saw it as SA owned thing that said, "use this for oauth".